src/utils/dom.js

```diff
@@ -55,17 +55,25 @@ export const style = (root, src) => new Promise((resolve, reject) => {
  * Creates a new Script DOM element
  * @param {Element} root Root node
  * @param {string} src Source
+ * @param {Object} [options={}] Options
  * @return {Promise<StyleElement>}
  */
-export const script = (root, src) => new Promise((resolve, reject) => {
+export const script = (root, src, options = {}) => new Promise((resolve, reject) => {
+  const opts = {
+    async: false,
+    defer: false,
+    onerror: (err) => reject(err),
```

Comment:
Maybe it would be better to hoist the `...options` a little bit here so that the callbacks can't be overridden ? That wold break the promise :thinking: 

So:

```
{
  async: false,
  defer: false,
  ...options,
  onerror: ...
  onload: ...
}
```

Reply:
I actually want to be able to run custom `onload` functions, that was the motivation behind this PR. The GAPI plugin needs a major update because Google did an overhaul of their APIs so I'm working on doing that on my end, and instead of the script functions being able to be chained with `then` statements they now have to use the `onload` attribute.

What about if I made it so that the custom `onload` function could be called before the resolve, like so:

```js
const opts = {
	async: false,
	defer: false,
	...options,
	onerror: (err) => {
		if (options.onerror) {
			options.onerror(err);
		}

		reject(err);
	},
	onload: () => {
		if (options.onload) {
			options.onload();
		}

		resolve(el);
	},
};
```

Would something like that be acceptable?

Reply:
Could you give me a usage example ?

Reply:
```js
const onload = () => createGapiClient(client);
return script($resourceRoot, src, { onload });
```

Reply:
I'm not sure I have enough context here, because this works in the exact same way:

```
script($resourceRoot, src).then(() => createGapiClient(client))
```

The difference is just a Promise VS Callback.